src/Fractions/Fraction.ConvertTo.cs

```diff
@@ -81,33 +87,167 @@ namespace Fractions;
+            return (decimal)numerator;
+        }
+
+        if (numerator > MAX_DECIMAL) {
+            if (denominator > MAX_DECIMAL) {
+                // both terms need to be rounded
+                numerator = (numerator * MAX_DECIMAL / denominator);
+                denominator = MAX_DECIMAL;
+                if (numerator <= MAX_DECIMAL) {
+                    // both terms are now within range
+                    return (decimal)numerator / (decimal)denominator;
+                }
+            }
+
+            var withoutDecimalPlaces = (decimal) BigInteger.DivRem(numerator, denominator, out var remainder);
+            return remainder.IsZero ? withoutDecimalPlaces : withoutDecimalPlaces + (decimal)remainder / (decimal)denominator;
+        }
+
+        if (numerator < MIN_DECIMAL) {
+            if (denominator > MAX_DECIMAL) {
+                // both terms need to be rounded
+                numerator = (numerator * MAX_DECIMAL / denominator);
+                denominator = MAX_DECIMAL;
+                if (numerator >= MIN_DECIMAL) {
+                    // both terms are now within range
+                    return (decimal)numerator / (decimal)denominator;
+                }
+            }
+
+            var withoutDecimalPlaces = (decimal)BigInteger.DivRem(numerator, denominator, out var remainder);
+            return remainder.IsZero ? withoutDecimalPlaces : withoutDecimalPlaces + (decimal)remainder / (decimal)denominator;
+        }
+
+        if (denominator > MAX_DECIMAL) {
+            // since both terms are non-zero and the numerator is smaller (in magnitude) to the denominator
+            // the resulting number would be in the range [-1,1] (exclusive)
+            // we want to flip the operation: x = a/b -> 1/x = b/a
+            var decimalPart = BigInteger.DivRem(denominator, numerator, out var remainder);
+            if (decimalPart < MAX_DECIMAL || decimalPart > MAX_DECIMAL) {
+                return decimal.Zero;
+            }
+
+            return remainder.IsZero ? 1m / (decimal)decimalPart : 1m / ((decimal)decimalPart + (decimal)remainder / (decimal)numerator);
+        }
+
+        return (decimal)numerator / (decimal)denominator;
+    }
+
+    /// <summary>
+    ///     Converts the fraction to a decimal number. If the fraction cannot be represented as a decimal due to its
+    ///     size, the method will return the closest possible decimal representation (either decimal.MaxValue or
+    ///     decimal.MinValue).
+    /// </summary>
+    /// <returns>
+    ///     The decimal representation of the fraction. If the fraction is too large to represent as a decimal, returns
+    ///     decimal.MaxValue. If the fraction is too small to represent as a decimal, returns decimal.MinValue.
+    /// </returns>
+    /// <remarks>If the fraction is NaN (the numerator and the denominator are both zero), the method will return decimal.Zero.</remarks>
+    public decimal ToDecimalSaturating() {

```

Comment:
Of course this one doesn't have to public, but I guess somebody might find some use for it (also, it was easier to test, rather than having to go through the INumber interface... 😄  ).

As a side note- a lot of the methods from `INumber` that are redundant with our existing properties (or are generally not interesting to the user) were implemented as explicit. Feel free to add or remove the methods that you feel would/ wouldn't be useful to the user (I'm talking about things like `IsFinite(Fraction)`, `IsInteger(Fraction)` etc..).